### ISSUE TITLE
Customizing service config for per-pod services

### ISSUE THREAD

**Issue by brownleej** [FoundationDB/fdb-kubernetes-operator#391]
We should have more options to configure the metadata and spec for the services we create when creating per-pod services. As part of this, we should explore what other service types we can handle.

**Comment by mengzhao-ibm** [comment]
Is this issue about adding support for Service type: NodePort, LoadBalancer, ExternalName? Any progress may I ask?

**Comment by mengzhao-ibm** [comment]
@ johscheuer Hi experts! 

I switched the sequence of addPods{} subReconciler and addServices{} subReconciler, because I want to pass pod’s zone information into service’s creation so need pod to be created first.

This brings up a problem. I need to update the already existed pod’s annotation(public-ip part) when I get the service ip. I used PodLifecycleManager.UpdateMetadata to update a pod annotation’s public-ip part and it worked, but I found it was not updated into sidecar env, so can’t be used for fdbmonitor.conf and fdb.cluster, which were still using pod.ip(the default value if there is no service ip).

I defined the downward API Volume File in pod’s spec.volumes.downwardAPI.items, and defined the the file in the container’s volumeMounts. It should be able to write a certain attribute of the pod, such as public-ip annotation, into the mounted volume. I did see this file which contains public ip value generated in the path inside of sidecar container that I’ve defined.

But the value of this attribute is dynamically updated. There is no file at the beginning, and it will be created and updated later.

My question is when to read this file into FDB_PUBLIC_IP parameter? I tried in getEnvForMonitorConfigSubstitution() in pod_models.go, but each time there is no file there, seem old environment has been cleaned up for each new reconciliation retry.
